dom.go

```diff
@@ -616,32 +624,35 @@ func (l keyedList) Key() interface{} {
 // context, unless keyed. Uses the currently known insertion point from the
 // parent to insert children at the correct position.
 func (l keyedList) reconcile(parent *HTML, prevChild ComponentOrHTML) (pendingMounts []Mounter) {
-	// Copy parent scope to render list elements into
+	// Effectively become the parent (copy its scope) so that we can reconcile
+	// our children against the prev child.
 	l.html.node = parent.node
 	l.html.insertBeforeNode = parent.insertBeforeNode
 	l.html.lastRenderedChild = parent.lastRenderedChild
+
 	switch v := prevChild.(type) {
 	case keyedList:
 		pendingMounts = l.html.reconcileChildren(v.html)
+	case nil:
```

Comment:
Is there any case where we might encounter a typed `nil` here? I wasn't certain, which is why I used the `nil` equality check in the default case instead.

Reply:
Nope. The `default` case isn't aware of the concrete type AND the actual `case` is not aware of the concrete type either. The behavior is 100% the same.

For example: https://play.golang.org/p/1t3iqGTfha

Reply:
But the behaviour needs to be different if the _value_ is nil.

Reply:
That is to say: your example code does the wrong thing here - it must to execute the `nil` case to produce the correct behaviour for this code.

Reply:
Maybe what you say is true, but that isn't handled today regardless of this change. This change doesn't change behavior at all. Does this mean there is already a bug here without this change?

For example, notice that `"caught nil!"` does not print in the default case here: https://play.golang.org/p/zvtbpHkeUJ

Reply:
Indeed it does.

Reply:
I'm not sure I follow? What is your "it does." comment in relationship to?

Reply:
Your question as to whether there was already a bug here.  I'll merge your changes, since I'm otherwise happy with them, and also fix this bug, then force-push the upstream PR.

Reply:
Oh okay, got it.